bleep-core/src/scala/bleep/commands/Test.scala

```diff
@@ -18,14 +40,62 @@ case class Test(watch: Boolean, projects: Array[model.CrossProjectName]) extends
   override def runWithServer(started: Started, bloop: BuildServer): Either[BleepException, Unit] =
     DoSourceGen(started, bloop, watchableProjects(started)).flatMap { _ =>
       val targets = BleepCommandRemote.buildTargets(started.buildPaths, projects)
-      val result = bloop.buildTargetTest(new bsp4j.TestParams(targets)).get()
-
-      result.getStatusCode match {
-        case bsp4j.StatusCode.OK =>
-          started.logger.info("Tests succeeded")
-          Right(())
-        case errorCode =>
-          Left(new BspCommandFailed("tests", projects, BspCommandFailed.StatusCode(errorCode)))
+      val tests = bloop.buildTargetScalaTestClasses(new bsp4j.ScalaTestClassesParams(targets)).get()
```

Comment:
ok, one other thing. can you adjust this method so that `buildTargetScalaTestClasses` is only ran if if one or both of `testSuitesOnly` or `testSuitesExclude` are set? It would be a shame to make the default case slower by calling bloop twice